File: lib/features/driver/data/models/emergency_report.dart
class EmergencyReport {
  final int id;
  final int driverId;
  final String driverName;
  final int? tripId;
  final String type;
  final String description;
  final String? location;
  final String status;
  final DateTime createdAt;

  const EmergencyReport({
    required this.id,
    required this.driverId,
    required this.driverName,
    this.tripId,
    required this.type,
    required this.description,
    this.location,
    required this.status,
    required this.createdAt,
  });

  factory EmergencyReport.fromJson(Map<String, dynamic> json) {
    return EmergencyReport(
      id: json['id'] as int,
      driverId: json['driverId'] as int,
      driverName: json['driverName'] as String? ?? 'Driver',
      tripId: json['tripId'] as int?,
      type: json['type'] as String? ?? 'EMERGENCY',
      description: json['description'] as String? ?? '',
      location: json['location'] as String?,
      status: json['status'] as String? ?? 'REPORTED',
      createdAt:
          DateTime.tryParse(json['createdAt'] as String? ?? '') ?? DateTime.now(),
    );
  }
}

File: backend/src/main/java/com/shuttle/domain/entity/EmergencyReport.java
package com.shuttle.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(
        name = "emergency_reports",
        indexes = {
                @Index(name = "idx_er_driver", columnList = "driver_id"),
                @Index(name = "idx_er_created", columnList = "created_at")
        }
)
public class EmergencyReport extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "driver_id", nullable = false)
    private Driver driver;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_id")
    private Trip trip;

    @Column(nullable = false, length = 50)
    private String type;

    @Column(nullable = false, length = 1000)
    private String description;

    @Column(length = 255)
    private String location;

    @Column(nullable = false, length = 32)
    private String status = "REPORTED";
}
